src/core/restrict-ifaces.c

```diff
@@ -0,0 +1,197 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <net/if.h>
+
+#if HAVE_LIBBPF
+#include <bpf/bpf.h>
+#include <bpf/libbpf.h>
+#endif
+
+#include "restrict-ifaces.h"
+
+#include "fd-util.h"
+#include "unit.h"
+
+#if BPF_FRAMEWORK
+/* libbpf, clang and llc compile time dependencies are satisfied */
+
+#include "bpf/restrict_ifaces/restrict-ifaces-skel.h"
+
+static struct restrict_ifaces_bpf *restrict_ifaces_bpf_free(struct restrict_ifaces_bpf *obj) {
+        if (obj && libbpf_get_error(obj) == 0)
+                (void) restrict_ifaces_bpf__destroy(obj);
+
+        return NULL;
+}
+
+DEFINE_TRIVIAL_CLEANUP_FUNC(struct restrict_ifaces_bpf *, restrict_ifaces_bpf_free);
+
+static int prepare_restrict_ifaces_bpf(Unit* u, bool is_allow_list,
+                const Set *restrict_network_interfaces,
+                struct restrict_ifaces_bpf **ret_object) {
+        _cleanup_(restrict_ifaces_bpf_freep) struct restrict_ifaces_bpf *obj = NULL;
+        char *iface;
+        uint8_t dummy = 0;
+        uint32_t ifindex;
+        int r, map_fd;
+
+        assert(ret_object);
+
+        obj = restrict_ifaces_bpf__open();
+        if (!obj)
+                return log_unit_error_errno(u, r, "Failed to open BPF object");
+
+        r = bpf_map__resize(obj->maps.ifaces_map, MAX(set_size(restrict_network_interfaces), 1u));
+        if (r != 0)
+                return log_unit_error_errno(u, r,
+                                "Failed to resize BPF map '%s': %m",
+                                bpf_map__name(obj->maps.ifaces_map));
+
+        obj->rodata->is_allow_list = (__u8) is_allow_list;
+
+        r = restrict_ifaces_bpf__load(obj);
+        if (r != 0)
+                return log_unit_error_errno(u, r, "Failed to load BPF object");
+
+        map_fd = bpf_map__fd(obj->maps.ifaces_map);
+        if (map_fd < 0)
+                return log_unit_error_errno(u, map_fd, "Failed to get BPF map fd");
+
+
+        /* Key zero indicates whether this is an allow or deny-list approach */
+        ifindex = 0;
+        dummy = (uint8_t) is_allow_list;
+        r = bpf_map_update_elem(map_fd, &ifindex, &dummy, BPF_ANY);
+        if (r < 0)
+                return log_unit_error_errno(u, r, "Failed to update BPF map '%s' fd: %m", map_name);
+        dummy = 0;
+
+        SET_FOREACH(iface, restrict_network_interfaces) {
+                ifindex = if_nametoindex(iface);
```

Comment:
Please use resolve_interface() from socket-netlink.c: it also supports resolving new-style "altnames" for interfaces

Reply:
Do you mean `resolve_ifname()`? `resolve_interface()` tries to parse the interface name to an integer before using netlink, I think it's not correct for this use case as an interface with an integer name can have a `if_index` different to the name.

Reply:
well, in networkd and elsewhere we generally explicitly check that people cannot name ifaces numerically, so that no ambiguity can happen, and so that ifindex and ifname can be used unambigiously interchangable. 

For transient services it might make sense to pinpoint ifaces by ifindex, since they are generated programmatically on the fly.  hence i see no eason why not to accept ifindexes here, if people want to specify them